### PR TITLE
Add `is_result` function

### LINKED ISSUE #365: Add `is_result` function
It should return `True` for `Result`, `FutureResult`. `RequiresContextResult`, `ContextResult`, `RequiresContextIOResult` and `ContextIOResult`

### PR DESCRIPTION
**Pull request by thepabloaguilar**
I've implemented the function, wrote tests, wrote examples in documentation.

Closes #365

### CODE REVIEW

**Comment by thepabloaguilar** [comment]
@sobolevn Can you take a look, please?

File: returns/instances.py [hunk 1]
```diff
@@ -17,3 +24,28 @@ def is_io(container: BaseContainer) -> bool:
...
+    result_containers = (
+        Result,
+        FutureResult,
+        ContextResult,
+        ContextIOResult,
+        IOResult,
+        RequiresContextResult,
+        RequiresContextIOResult,
+    )
+    return isinstance(container, result_containers)
```
Comment (amaksymov) [inline returns/instances.py:51]: I have some nit.
I think a better name for the variable is `result_container_type` than `result_containers` and in this case maybe the best not to use a temporary variable for one call and just use:
```
  return isinstance(container, (
    Result,
    FutureResult,
    ContextResult,
    ContextIOResult,
    IOResult,
    RequiresContextResult,
    RequiresContextIOResult,
  ))
```
Sorry, if this remark is meaningless.

Comment (thepabloaguilar) [reply returns/instances.py:51]: I agree with the variable's name, but not with removing the variable!
Using the variable it's more readable than passing all possible types to `isinstance` function.

**Comment by thepabloaguilar** [comment]
I'll close this PT 'cause that functionality is not required so far!

**Pull request closed without merging**